### ISSUE TITLE
[Feature Request] Support Switchres using the downsampling option (Libretro)

### ISSUE THREAD

**Issue by DukeSkinny** [FCare/Kronos#1105]
First of all, thank you for implementing the downsampling option. My request is to improve upon that by optionally allowing the emulator to report the native resolution even when the internal resolution is increased. This is helpful when using Switchres in Retroarch.

**Comment by barbudreadmon** [comment]
That's what the downsampling option is about.

**Comment by DukeSkinny** [comment]
> 
> 
> That's what the downsampling option is about.

I understood it as such, but then came to see the implementation as working a bit differently. If that is the case, however, it doesn't work properly. If it did, Switchres would correctly switch on a 2x internal resolution downsampled to 240p.

Am I wrong to assume this hasn't been tested on a CRT monitor? Also, please let me know if there is any information in particular I can provide.

**Edit** [Here](https://pastebin.com/aiBhzHRD) is the log from running Bulk Slash using 2x. Note that the game normally switches from 224 to 448 on the title screen, and then back to 224 in-game.

**Comment by barbudreadmon** [comment]
Again, "Output to original resolution" is doing exactly what you are requesting here, nothing less nothing more. 
Sorry but i don't have a crt monitor and i don't know what's required for your switchres thingy. Tell me and maybe i'll be able to figure out something.

**Comment by DukeSkinny** [comment]
Yes, and as much can be verified in RA. The resolution reported there is native, which makes the issue difficult to understand. Normally, Switchres correctly switches resolution if 2 conditions are met:

1. The reported resolution is within the range of a CRT monitor
2. The requested resolution (or a predefined multiple on the horizontal axis) is among the enumerated resolutions in the monitor EDID

In this case both are true, yet the CRT's output resolution is stuck at 640x448, corresponding to either the Bulk Slash menu resolution or the general game resolution before downsampling occurs.
I can try getting in touch with Ben Templeman who maintains Switchres for RA.

**Comment by barbudreadmon** [comment]
>I can try getting in touch with Ben Templeman who maintains Switchres for RA.

Yes, getting in touch with someone knowledgeable about this sounds like a good idea
